### PR TITLE
Matthew Fix

### PR DESCRIPTION
**Pull request by jonpsy**
# Description

Brief Description of the PR:

Fixes # (issue)
#2339 

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [x] Additional Testing

# Checklist:

- [ x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

The issue created by the user is perfectly solved by this

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:

### CODE REVIEW

**Comment by jonpsy** [comment]
This should work for one hot encoded label. I'll work on sparse labels as well but I wanted to gather some reviews before. Looking forward! :)

**Comment by bhack** [comment]
@ZeynepP Can you test this fix?

**Comment by jonpsy** [comment]
@bhack  I've added test case which confirms that patch solves the issue. Would you mind reviewing this patch?

**Comment by bhack** [comment]
Can we check numeric with https://scikit-learn.org/stable/modules/generated/sklearn.metrics.matthews_corrcoef.html ?

**Comment by jonpsy** [comment]
Its in the test case :)

**Comment by bhack** [comment]
>  I should import sklearn and check their output values? I can do that.
Yes we already have some tests with `sklearn` in the repo. I think that you can check these.

**Comment by jonpsy** [comment]
Done :)

File: tensorflow_addons/metrics/matthews_correlation_coefficient.py [hunk 1]
```diff
@@ -70,73 +70,47 @@ def __init__(
...
-        self.true_negatives = self.add_weight(
-            "true_negatives",
-            shape=[self.num_classes],
-            initializer="zeros",
-            dtype=self.dtype,
+        self.conf_mtx = self.add_weight(
+            "conf_mtx",
+            shape=(self.num_classes, self.num_classes),
+            initializer=tf.keras.initializers.zeros,
+            dtype=tf.float32,
```
Comment (bhack) [inline tensorflow_addons/metrics/matthews_correlation_coefficient.py:77]: is this self.dtype?

**Review by bhack — APPROVED** [review]

File: tensorflow_addons/metrics/tests/matthews_correlation_coefficient_test.py [hunk 2]
```diff
@@ -36,30 +37,59 @@ def check_results(obj, value):
...
-    preds = tf.constant([[1.0], [0.0], [1.0], [1.0]], dtype=tf.float32)
+    gt_label = tf.constant(
+        [[0.0, 1.0], [0.0, 1.0], [0.0, 1.0], [1.0, 0.0]], dtype=tf.float32
+    )
+    preds = tf.constant(
+        [[0.0, 1.0], [1.0, 0.0], [0.0, 1.0], [0.0, 1.0]], dtype=tf.float32
+    )
     # Initialize
-    mcc = MatthewsCorrelationCoefficient(1)
+    mcc = MatthewsCorrelationCoefficient(2)
```
Comment (lgeiger) [inline tensorflow_addons/metrics/tests/matthews_correlation_coefficient_test.py:47]: Unfortunately this PR breaks this test and the use of MCC with `num_classes=2`. Is this breaking change intended?

Comment (jonpsy) [reply tensorflow_addons/metrics/tests/matthews_correlation_coefficient_test.py:47]: Let's continue this discussion on your PR.
